compiler/src/iree/compiler/Dialect/VM/Conversion/VMToEmitC/ConvertVMToEmitC.cpp

```diff
@@ -1352,7 +1352,8 @@ class ExportOpConversion : public OpConversionPattern<IREE::VM::ExportOp> {
     auto loc = exportOp.getLoc();
 
     IREE::VM::EmitCTypeConverter *typeConverter =
-        this->getTypeConverter<IREE::VM::EmitCTypeConverter>();
+        const_cast<IREE::VM::EmitCTypeConverter *>(
+            getTypeConverter<const IREE::VM::EmitCTypeConverter>());
```

Comment:
I have bad fixes in this file because the type converter has a map variable for caching. We can't unconditionally add `const` to all the method. Otherwise, it would say that `DenseMap::insert` method is not a const method.

Reply:
@simon-camp @benvanik could you take a look if they are acceptable fixes?

Reply:
Could you point me to which line results in an error if this `const_cast` was dropped.

Reply:
Here are some examples I had issues:

l.1361:

```
auto vmAnalysis = typeConverter->lookupAnalysis(funcOp);
```

l.1395:

```
    typeConverter->analysisCache.insert(
        std::make_pair(newFuncOp.getOperation(), VMAnalysis(functionType)));
```

The former example requires [all methods ](https://github.com/openxla/iree/blob/aef8d3ce065826e915a45e12135d6f791737d8b4/compiler/src/iree/compiler/Dialect/VM/Conversion/VMToEmitC/EmitCTypeConverter.h#L22-L60)to be added `const` qualified. This is very tricky for [lookupAnalysis](https://github.com/openxla/iree/blob/aef8d3ce065826e915a45e12135d6f791737d8b4/compiler/src/iree/compiler/Dialect/VM/Conversion/VMToEmitC/EmitCTypeConverter.cpp#L119-L127) method because the infer-ed type mismatch return type.

The latter example is very very very tricky. `analysisCache` has [llvm::DenseMap<Type, int> type](https://github.com/openxla/iree/blob/aef8d3ce065826e915a45e12135d6f791737d8b4/compiler/src/iree/compiler/Dialect/VM/Conversion/VMToEmitC/EmitCTypeConverter.h#L58), and it will require `insert` method having `const` qualified -- which is impossible to me.

Reply:
Thanks, will do!

Reply:
https://github.com/openxla/iree/issues/14702 filed. I'm not sure why I can't assign it to you..